app/src/main/java/com/eveningoutpost/dexdrip/Models/blueReader.java

```diff
@@ -159,9 +165,10 @@ private static synchronized void processNewTransmitterData(TransmitterData trans
             PersistentStore.setLong("blueReader_Full_Battery", transmitterData.sensor_battery_level);
             Log.i(TAG, "blueReader_Full_Battery set to: " + transmitterData.sensor_battery_level) ;
         }
-        Pref.setInt("bridge_battery", ((transmitterData.sensor_battery_level - 3300) * 100 / (((int) (long) PersistentStore.getLong("blueReader_Full_Battery"))-3300)));
-        sensor.latest_battery_level = ((transmitterData.sensor_battery_level - 3300) * 100 / (((int) (long) PersistentStore.getLong("blueReader_Full_Battery"))-3300));
-        blueReaderDays = 6.129200670865791d / (1d + Math.pow(((double)transmitterData.sensor_battery_level/3763.700630306379d),(-61.04241888028577d)));
+        int localBridgeBattery =((transmitterData.sensor_battery_level - 3300) * 100 / (((int) (long) PersistentStore.getLong("blueReader_Full_Battery"))-3300));
+        Pref.setInt("bridge_battery", localBridgeBattery);
+        sensor.latest_battery_level = localBridgeBattery;
+        blueReaderDays = 6.129200670865791d / (1d + Math.pow(((double)transmitterData.sensor_battery_level/3763.700630306379d),(-61.04241888028577d))); //todo compare with test-formular, and new Data of batterylog
```

Comment:
As mentioned, I would really like to know the background for this formula, but it is ok to do that in a later pull-request.

Reply:
Yes will do that later, as I have a lot of new data. The formula itself (to answer your question) comes from a regession method I have used.

Reply:
Hi. As promised here the sequence of info given about BlueReader battery % and days Left:


   Date |       Time      | Battery %|  Life Days
06-apr |     8:00 P.M. |       83      | 6.0
08-apr |     8:00 P.M. |       62      | 4.8
09-apr |     8:00 P.M. |       58      | 3.9
11-apr |     8:00 P.M. |       48      | 1.9
12-apr |     8:00 P.M. |       41      | 0.9
13-apr |     8:00 P.M. |       37      | 0.6
14-apr |   12:00 A.M. |       21      | 0.1
14-apr |     1:00 P.M. |      -40      | 0.1 (DEAD!)
                                        |
                          (yes, it's minus 40!)

This is the resulting graph:
![image](https://user-images.githubusercontent.com/32884006/38735202-6bbbb068-3f28-11e8-918f-df691479b159.png)

BTW, I have a setting that notifies when BlueReader battery goes below 20%.
Hope this helps.
Bye.
Pippo   :-)